drivers/gles3/shaders/canvas.glsl

```diff
@@ -502,6 +502,8 @@ float map_ninepatch_axis(float pixel, float draw_size, float tex_pixel_size, flo
 			ratio = mod(ratio * scale, 1.0);
 			// Scale to source texture.
 			return (margin_begin + ratio * dst_area) * tex_pixel_size;
+		} else if (np_repeat == 3) { // Center.
+			return clamp(pixel - (draw_size - tex_size) / 2.0, margin_begin, tex_size - margin_end) * tex_pixel_size;
```

Comment:
Straight up removal of that `+1`/`-1` offset for sampling is not good too, in the end we want to sample from the center in that stretched part I guess... So probably using some epsilon for this should do the thing? :thinking:

| texture | current version | suggested with `eps` |
|--------|--------|--------|
|![dev x86_64_UMIaL8iXhS](https://github.com/godotengine/godot/assets/9283098/c6783562-27e0-44f1-bc95-431cad094443)|![mcUoxKD9CE](https://github.com/godotengine/godot/assets/9283098/4f55d530-b7b5-4841-b87d-2c0b9b43c5e3)|![3w3ElgCxLl](https://github.com/godotengine/godot/assets/9283098/edd70f9c-1c64-48ee-86a4-f1af8acfaa94)|
||![23gRRdc189](https://github.com/godotengine/godot/assets/9283098/c7aec2b4-f1a6-487d-aaf7-f18ac6828617)|![6p4iuLQDIa](https://github.com/godotengine/godot/assets/9283098/2fba7e04-7fd7-4a20-a299-6fd3253aa336)|



```suggestion
			const float eps = 1e-5;
			return clamp(pixel - (draw_size - tex_size) / 2.0, margin_begin + eps, tex_size - margin_end - eps) * tex_pixel_size;
```

Reply:
Use `CMP_EPSILON` instead of `float eps`.

Reply:
There's no CMP_EPSILON available in the GLSL file _(I also tried it for some reason and it didn't work)_, so this may have to do.

Reply:
Apologies, I missed that this was GLSL.